tests/spec/core/dfn-abbr-spec.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+describe("Core — Definition Abbreviations", () => {
+  afterAll(flushIframes);
+  it("processes definition abbreviations", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `
+      <section id="section">
+        <dfn data-abbr>foo bar</dfn> can be referenced as
+        <span id="foobarLink">
+          <a id="dfnlink">FB</a>
+        </span>
+      </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const dfnFooBar = doc.querySelector("#section dfn");
+    expect(dfnFooBar.id).toEqual("dfn-fb");
+    expect(dfnFooBar.dataset.abbr).toEqual("FB");
+    expect(dfnFooBar.textContent.trim()).toEqual("foo bar (FB)");
+    const linkToDfn = doc.getElementById("dfnlink");
+    expect(linkToDfn.getAttribute("href")).toEqual("#dfn-fb");
+    expect(linkToDfn.querySelector("abbr").textContent.trim()).toEqual("FB");
+  });
+  it("allows different abbreviation combinations to link to dfn", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `
+      <section id="section">
+        <dfn data-abbr>foo bar</dfn> can be referenced as
+        <span id="foobarLink">
+          <a class="dfnlink">FB</a>
+          <a class="dfnlink">foo bar</a>
+          <a class="dfnlink">foo bar (FB)</a>
+        </span>
+      </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const dfnFooBar = doc.querySelector("#section dfn");
+    expect(dfnFooBar.id).toEqual("dfn-fb");
+    expect(dfnFooBar.dataset.abbr).toEqual("FB");
+    expect(dfnFooBar.textContent.trim()).toEqual("foo bar (FB)");
+    const linkToDfn = doc.querySelectorAll(".dfnlink");
+    linkToDfn.forEach(link =>
+      expect(link.getAttribute("href")).toEqual("#dfn-fb")
+    );
+  });
+  it("correctly parses and abbreviates different Abbreviation syntax", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `
+      <section id="section">
+        <dfn data-abbr>United States</dfn>
+        <dfn data-abbr="PoR">Position of Responsibility</dfn>
+        <dfn data-abbr="UI">User Interface</dfn>
+        <div>US</div>
+        <div>PoR</div>
+        <div>UI</div>
+      </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+    const divs = doc.querySelectorAll("#section div");
```

Comment:
```suggestion
    const divs = [...doc.querySelectorAll("#section div")];
```

Reply:
Why? We are not using any Array-specific functions here.

Reply:
Oh true. I forgot NodeList has .forEach(). I guess only for destructoring below.

Reply:
NodeList already supports destructuring so I think `dfns` neither have to be an array.

Reply:
TIL! Thanks @saschanaz.